modules/videoio/src/cap_avi_container.cpp

```diff
@@ -0,0 +1,799 @@
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+
+#include "cap_avi_container.hpp"
+
+namespace cv
+{
+
+String fourccToString(uint32_t fourcc)
+{
+    return format("%c%c%c%c", fourcc & 255, (fourcc >> 8) & 255, (fourcc >> 16) & 255, (fourcc >> 24) & 255);
+}
+
+MjpegInputStream::MjpegInputStream(): m_is_valid(false), m_f(0)
+{
+}
+
+MjpegInputStream::MjpegInputStream(const String& filename): m_is_valid(false), m_f(0)
+{
+    open(filename);
+}
+
+bool MjpegInputStream::isOpened() const
+{
+    return m_f != 0;
+}
+
+bool MjpegInputStream::open(const String& filename)
+{
+    close();
+
+    m_f = fopen(filename.c_str(), "rb");
+
+    m_is_valid = isOpened();
+
+    return m_is_valid;
+}
+
+void MjpegInputStream::close()
+{
+    if(isOpened())
+    {
+        m_is_valid = false;
+
+        fclose(m_f);
+        m_f = 0;
+    }
+}
+
+MjpegInputStream& MjpegInputStream::read(char* buf, uint64_t count)
+{
+    if(isOpened())
+    {
+        m_is_valid = (count == fread((void*)buf, 1, (size_t)count, m_f));
+    }
+
+    return *this;
+}
+
+MjpegInputStream& MjpegInputStream::seekg(uint64_t pos)
+{
+    m_is_valid = (fseek(m_f, (long)pos, SEEK_SET) == 0);
+
+    return *this;
+}
+
+uint64_t MjpegInputStream::tellg()
+{
+    return ftell(m_f);
+}
+
+MjpegInputStream::operator bool()
+{
+    return m_is_valid;
+}
+
+MjpegInputStream::~MjpegInputStream()
+{
+    close();
+}
+
+bool AVIReadContainer::parseIndex(uint32_t index_size, frame_list& in_frame_list)
+{
+    uint64_t index_end = in_str->tellg();
+    index_end += index_size;
+    bool result = false;
+
+    while(in_str && (in_str->tellg() < index_end))
+    {
+        AviIndex idx1;
+        *in_str >> idx1;
+
+        if(idx1.ckid == m_stream_id)
+        {
+            uint64_t absolute_pos = m_movi_start + idx1.dwChunkOffset;
+
+            if(absolute_pos < m_movi_end)
+            {
+                in_frame_list.push_back(std::make_pair(absolute_pos, idx1.dwChunkLength));
+            }
+            else
+            {
+                //unsupported case
+                fprintf(stderr, "Frame offset points outside movi section.\n");
+            }
+        }
+
+        result = true;
+    }
+
+    return result;
+}
+
+bool AVIReadContainer::parseStrl(uint8_t stream_id)
+{
+    RiffChunk strh;
+    *in_str >> strh;
+
+    if(in_str && strh.m_four_cc == STRH_CC)
+    {
+        uint64_t next_strl_list = in_str->tellg();
+        next_strl_list += strh.m_size;
+
+        AviStreamHeader strm_hdr;
+        *in_str >> strm_hdr;
+
+        if(strm_hdr.fccType == VIDS_CC && strm_hdr.fccHandler == MJPG_CC)
```

Comment:
Probably we should pass FourCC code (_MJPG_CC_ here) as a parameter into this method.

Reply:
@mshabunin corrected